### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by dbjsdev**
https://trello.com/c/AWmbz8Vu/4-19-parent-and-subsidiaries-permissions

Early PR for review before insertion of hardcoded permissions data

### CODE REVIEW

File: server/models/cache/singletons/permissions.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  constructor() {}
+
+  static async initialize() {
+
+  }
+
+  static myPermission(dataOwner, dataPermissions, role){
+    return ALL_PERMISSIONS
+        .filter(x => 
+            (!x.dataOwner.length || dataOwner.toLowerCase() == x.dataOwner.toLowerCase()) &&
```
Comment (wozzer72) [inline server/models/cache/singletons/permissions.js]: These are an "eeither, eiither, or" not an "and".

Comment (wozzer72) [reply server/models/cache/singletons/permissions.js]: Also, you're only interested in checking data permissions if dataOwner is not `parent`

File: server/models/cache/singletons/permissions.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+];
+
+class PermissionCache {
+  constructor() {}
+
+  static async initialize() {
+
+  }
+
+  static myPermission(dataOwner, dataPermissions, role){
```
Comment (wozzer72) [inline server/models/cache/singletons/permissions.js]: Should this be "myPermissions" (plural - because it's returning many permissions

Comment (dbjsdev) [reply server/models/cache/singletons/permissions.js]: Corrected

File: server/models/cache/singletons/permissions.js (hunk 1 above)
Comment (dbjsdev) [reply server/models/cache/singletons/permissions.js]: Made an attempt at this

**Pull request merged**